File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/main/MainActivity.java
package de.phillippbertram.cleanandroid.main;

import android.os.Bundle;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;
import de.phillippbertram.cleanandroid.R;


public class MainActivity extends RxAppCompatActivity implements MainView {

    private final MainPresenter presenter = new MainPresenter();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        presenter.bind(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.unbind();
    }
}

File: CleanAndroid/app/src/main/java/de/phillippbertram/cleanandroid/main/MainActivity.kt
package de.phillippbertram.cleanandroid.main

import android.content.Intent
import android.os.Bundle
import butterknife.ButterKnife
import butterknife.OnClick
import de.phillippbertram.cleanandroid.R
import de.phillippbertram.cleanandroid.chat.ChatListActivity
import de.phillippbertram.cleanandroid.internal.base.RxActivity

open class MainActivity : RxActivity<MainView, MainPresenter>() {

    override fun createPresenter(): MainPresenter {
        return MainPresenter()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        ButterKnife.bind(this)
    }

    @OnClick(R.id.detailButton)
    open fun onDetailButtonClicked() {
        val intent = Intent(this, ChatListActivity::class.java)
        startActivity(intent)
    }
}
